syntax/printer.go

```diff
@@ -1081,6 +1081,13 @@ func (p *Printer) command(cmd Command, redirs []*Redirect) (startRedirs int) {
 		p.WriteByte('(')
 		if len(x.Stmts) > 0 && startsWithLparen(x.Stmts[0]) {
 			p.wantSpace = spaceRequired
+			if nested, ok := x.Stmts[0].Cmd.(*Subshell); ok {
+				// we only want to keep the space between two nested subshells' open brackets
+				// if its in a single line to avoid ambiguity
+				if x.Lparen.Line() != nested.Pos().Line() {
```

Comment:
hmm, I'm not sure if this is exactly right - in some cases we'll add a newline, like:

```
$ cat f.sh
( (foo); bar)
$ shfmt f.sh | tr " \t" "*?"
(*
?(foo)
?bar
)
```

You can see that we add a space at the end of the line, which I replaced with `*` to make it visible. I think this edge case is still broken with your fix.

Reply:
thanks for the tip! I added some more scenarios and test cases.